python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -111,6 +120,84 @@ cdef class Scalar:
         """
         return _from_py(py_val)
 
+    @classmethod
+    def from_numpy(cls, numpy_val):
+        """
+        Convert a NumPy scalar to a Scalar.
+
+        Parameters
+        ----------
+        numpy_val: numpy.generic
+            Value to convert to a pylibcudf.Scalar
+
+        Returns
+        -------
+        Scalar
+            New pylibcudf.Scalar
+        """
+        cdef DataType dtype
+        cdef unique_ptr[scalar] c_obj
+        cdef Scalar slr
+
+        import numpy as np
```

Comment:
Can you remind me why we're locally importing numpy here?

Reply:
I believe we want pylibcudf to have a minimal set of required dependencies and therefore not have numpy as a required dependency.

I could do a 

```python
try:
    import numpy as np
    has_numpy = True
except ImportError:
    has_numpy = False
```

At the top and then check `has_numpy` in the method if that's preferable

Reply:
> I believe we want pylibcudf to have a minimal set of required dependencies and therefore not have numpy as a required dependency.

Ah ok makes sense. I think the only case where there might be a problem is if something tries to pass this function a non-numpy object without numpy being installed. In that case, I think we'll hit a `ModuleNotFoundError` here. Maybe that's informative enough, but since it's at the level of an external facing API maybe its worth adding some detail. 

In any event it probably is a bit better to figure it out once at the top of the file, if we use this in a tight loop it ends up being just a check against a bool instead of whatever python is doing to `import` even something cached again.

Reply:
I agree with Brandon here.